### ISSUE TITLE
App fails to connect to DeepL server if using Private DNS with ad blocking

### ISSUE THREAD

**Issue by erazemk** [sakusaku3939/DeepLAndroid#13]
When Private DNS (Android 9+) is enabled and connected to an ad blocking server (e.g. dns.adguard.com), the app cannot connect to the DeepL server, even though the DeepL website works fine if it's enabled. I have not tested with other DNS providers, but the problem might affect them too.

I am using Firefox as my browser, so the engine isn't WebWiew, the problem might be there or in the way the app connects to the server.

If needed I can provide a logcat.

To recreate:
* Enable Private DNS (Use `dns.adguard.com` as an example)
* Open the app (force stop it previously if necessary)
* The app won't connect to the server
* Disable Private DNS and force stop the app
* Open the app
* It will connect properly

**Comment by najodleglejszy** [comment]
I checked my NextDNS logs and it seems that `s.deepl.com` is being blocked. if I whitelist it, it connects without issues. no idea what does that URL do though, or why it's not an issue when connecting to the website via Firefox.

**Comment by BluePixel4k** [comment]
I have the same problem.
I think that s.deepl.com is used for the statistics.

**Comment by erazemk** [comment]
Could it be that the app stops trying to connect if one of the connections fails?

**Comment by peterge1998** [comment]
I am using Adaway and it also doesnt allow the connection in this app

**Comment by sakusaku3939** [comment]
I was able to reproduce the error, so I investigated the logs.
The reason was that the sending process to the DAP server failed only when Adguard DNS was used, and onReceivedError was called.
However, since there was no problem with the connection to the DeepL server itself, the problem was solved by modifying the code on the application side.

https://github.com/sakusaku3939/DeepLAndroid/releases/tag/v3.5
